scheduler/src/cook/mesos/rebalancer.clj

```diff
@@ -155,10 +155,35 @@
   [dru]
   (* dru metrics-dru-scale))
 
-(defn compute-pending-job-dru
+(defn compute-pending-gpu-job-dru
+  "Takes state and a pending gpu job entity, returns the dru of the pending-job.
+
+   This algorithm only should be used on jobs that use GPUs. It computes the GPU DRU."
+  [{:keys [task->scored-task user->sorted-running-task-ents user->dru-divisors] :as state}
+   pending-job-ent]
+  (let [user (:job/user pending-job-ent)
+        {gpu-req :gpus} (util/job-ent->resources pending-job-ent)
+        gpu-divisor (get-in user->dru-divisors [user :gpus])
+        pending-task-ent (util/create-task-ent pending-job-ent)
+        nearest-task-ent (some-> user->sorted-running-task-ents
+                                 (get user)
+                                 (rsubseq <= pending-task-ent)
+                                 (first))
+        nearest-task-dru (if nearest-task-ent
+                           (get task->scored-task nearest-task-ent)
+                           0.0)
+        pending-job-dru (+ nearest-task-dru (/ gpu-req gpu-divisor))]
```

Comment:
That is a great question! I copied this from the previous code, so I had to reverse engineer what's going on here.

The goal of this function is to compute the DRU of a pending job w/o redoing the whole DRU sorting process. So, we first get the `nearest-task-dru`. This is lookup into the sorted set of the user's tasks (since we're computing this on a per-user basis). We find the DRU of the highest priority task for that user that's adjacent to the new pending task's priority.

Next, we add the marginal DRU contribution of the pending task to the adjacent task, since remember, DRU is cumulative.